components/layout/layout.directive.js

```diff
@@ -1,64 +1,126 @@
-export default function (HsCore, $timeout, HsConfig, $compile, HsLayoutService) {
+/**
+ * @param HsCore
+ * @param $timeout
+ * @param HsConfig
+ * @param $compile
+ * @param HsLayoutService
+ */
+export default function (
+  HsCore,
+  $timeout,
+  HsConfig,
+  $compile,
+  HsLayoutService
+) {
   'ngInject';
   return {
-	template: HsConfig.design == 'md' ?
-		HsConfig.directiveTemplates.layout ||
-		require('./partials/layoutmd.html') :
-		require('./partials/layout.html'),
+    template:
+      HsConfig.design == 'md'
+        ? HsConfig.directiveTemplates.layout ||
+          require('./partials/layoutmd.html')
+        : require('./partials/layout.html'),
     link: function (scope, element) {
-      HsLayoutService.contentWrapper = element[0].querySelector('.hs-content-wrapper');
+      HsLayoutService.contentWrapper = element[0].querySelector(
+        '.hs-content-wrapper'
+      );
       HsLayoutService.layoutElement = element[0];
 
+      setTimeout(() => {
+        const hsapp = document.getElementById('hs-app');
```

Comment:
We should not rely on this id to be set. There can also be multiple hs elements on the page. I propose to just climb back from linked element using parent. It should even be possible in ng9 using components elementRef